modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -305,6 +309,27 @@ public static boolean isVoid(Method method)
         return getAllMethods(declaringClass, withAnnotation(Operation.class), withModifier(Modifier.PUBLIC));
     }
 
+    public static Collection<Field> getExposedFields(Class<?> extensionType)
+    {
+        Collection<Field> allFields = getParameterFields(extensionType);
+        if (!allFields.isEmpty())
+        {
+            return allFields;
+        }
+        try
+        {
+            PropertyDescriptor[] propertyDescriptors = Introspector.getBeanInfo(extensionType, Object.class).getPropertyDescriptors();
+            return Arrays.stream(propertyDescriptors)
+                    .filter(propertyDescriptor -> propertyDescriptor.getReadMethod() != null)
+                    .map(propertyDescriptor -> getField(extensionType, propertyDescriptor.getName(), propertyDescriptor.getPropertyType()))
```

Comment:
add a test case which verifies that field will never be null, even if the getter is not backed by a field of a consistent name